custom/icds_reports/sqldata/exports/aww_activity_report.py

```diff
@@ -0,0 +1,80 @@
+from corehq.apps.locations.models import SQLLocation
+from custom.icds_reports.models.aggregate import AggregateInactiveAWW
+from custom.icds_reports.utils import india_now, DATA_NOT_ENTERED
+
+
+class AwwActivityExport(object):
+    title = 'AWW Activity Report'
+
+    def __init__(self, config, loc_level=0, beta=False):
+        self.config = config
+        self.loc_level = loc_level
+        self.beta = beta
+
+    def get_excel_data(self, location):
+
+        def _format_infrastructure_data(data):
+            return data if data is not None else DATA_NOT_ENTERED
+
+        def _format_date(data):
+            return data.strftime("%d-%m-%Y") if data is not DATA_NOT_ENTERED else data
+
+        filters = {}
+
+        if self.loc_level == 4:
```

Comment:
this will result in filter dict like 
```
{'<location>_id':<location_id>}

``` 

yeah?

I think filter should also include all the parent locations as well. It would improve the cost of the the query. For example:
if you will query for filters = {block_id:23456678}. It will look for that block in all the states. So good to have all the ancestors in the filter.

Reply:
updated